src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -856,5 +858,34 @@ def disk_configurations(vm_type, existing_disk_cids=[])
       )
       disk_configurations.push(ephemeral_disk_config)
     end
+
+
+    # return name info as a hash
+    # if  there are both instance group name and deployment name
+    #   return a valid hash name info table
+    # else return a empty hash
+    #
+    # The bosh is guaranteed to give a environment with no error
+    # which means in the groups array , there are always
+    # director-name, deployment-name and job-name in order
+
+    def get_name_info_from_bosh_env(environment)
+      name_info = {}
+      if environment.nil? || environment.empty?
+        return name_info
+      end
+      instance_group_name = environment.fetch('bosh', {}).fetch('groups', {}).fetch(2,"")
```

Comment:
You can use `Hash#dig` here.

http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/